platform/kube/admit/admit.go

```diff
@@ -0,0 +1,248 @@
+// pilot configuration.
+type Admit struct {
+	descriptor       model.ConfigDescriptor
+	hookConfigName   string
+	hookName         string
+	serviceName      string
+	serviceNamespace string
+	caBundle         []byte
+
+	// unconditionally validate all config that is not in this list of
+	// configuration.
+	validateNamespaces []string
+}
+
+// GetAPIServerExtensionCACert gets the CA cert that will signed the
+// cert used by the "GenericAdmissionWebhook" plugin admission
+// controller.
+func GetAPIServerExtensionCACert(cl kubernetes.Interface) ([]byte, error) {
+	const name = "extension-apiserver-authentication"
+	c, err := cl.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	pem, ok := c.Data["requestheader-client-ca-file"]
+	if !ok {
+		return nil, fmt.Errorf("cannot find ca.crt in %v: ConfigMap.Data is %#v", name, c.Data)
+	}
+	return []byte(pem), nil
+}
+
+// MakeTLSConfig makes a TLS configuration suitable for use with the
+// GenericAdmissionWebhook.
+func MakeTLSConfig(serverCert, serverKey, caCert []byte) (*tls.Config, error) {
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+	cert, err := tls.X509KeyPair(serverCert, serverKey)
+	if err != nil {
+		return nil, err
+	}
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		ClientCAs:    caCertPool,
+		ClientAuth:   tls.RequireAndVerifyClientCert,
+	}, nil
+}
+
+// New creates a new instance of the admission webhook controller.
+func New(descriptor model.ConfigDescriptor, hookConfigName, hookName, serviceName, serviceNamespace string, validateNamespaces []string, caBundle []byte) *Admit { // nolint: lll
+	return &Admit{
+		descriptor:         descriptor,
+		hookConfigName:     hookConfigName,
+		hookName:           hookName,
+		serviceName:        serviceName,
+		serviceNamespace:   serviceNamespace,
+		caBundle:           caBundle,
+		validateNamespaces: validateNamespaces,
+	}
+}
+
+// Unregister registers the external admission webhook
+func (a *Admit) Unregister(client admissionClient.ExternalAdmissionHookConfigurationInterface) error {
+	return client.Delete(a.hookConfigName, nil)
+}
+
+// Register registers the external admission webhook for pilot
+// configuration types.
+func (a *Admit) Register(client admissionClient.ExternalAdmissionHookConfigurationInterface) error {
+	var resources []string
+	for _, schema := range a.descriptor {
+		resources = append(resources, schema.Plural)
+	}
+
+	webhook := &admissionregistrationv1alpha1.ExternalAdmissionHookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: a.hookConfigName,
+		},
+		ExternalAdmissionHooks: []admissionregistrationv1alpha1.ExternalAdmissionHook{
+			{
+				Name: a.hookName,
+				Rules: []admissionregistrationv1alpha1.RuleWithOperations{{
+					Operations: []admissionregistrationv1alpha1.OperationType{
+						admissionregistrationv1alpha1.Create,
+						admissionregistrationv1alpha1.Update,
```

Comment:
PATCH?

Reply:
There is no `PATCH` operation for admission (see https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/admissionregistration/types.go#L186